src/library/qasm_new_to_old.hpp

```diff
@@ -92,6 +93,13 @@ static Operation *convert_instruction(const cqasm::semantic::Instruction &instru
                 Qubits(convert_indices(instruction.operands[0]->as_qubit_refs()->index))
             );
             break;
+        case ParameterType::SingleQubitInt:
+            op = new Operation(
+                instruction.instruction->name,
+                instruction.operands[1]->as_const_int()->value,
```

Comment:
```suggestion
                (const int)instruction.operands[1]->as_const_int()->value,
```
to solve the ambiguity.
Also this parameter should be the third parameter.

Reply:
Solved ambiguity